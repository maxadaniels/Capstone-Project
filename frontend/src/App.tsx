import { useEffect, useState } from 'react';
import { Container } from 'react-bootstrap';
import { Route, Routes } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import LoginModal from './components/LoginModal';
import NavBar from './components/NavBar';
import SignUpModal from './components/SignUpModal';
import { User } from './models/user';
import * as NotesApi from "./network/notes_api";
import NotesPage from './pages/NotesPage';
import NotFoundPage from './pages/NotFoundPage';
import PrivacyPage from './pages/PrivacyPage';
import styles from "./styles/App.module.css";

function App() {
    // State to store the currently logged-in user
	const [loggedInUser, setLoggedInUser] = useState<User | null>(null);

	// State to control the visibility of the sign-up modal
	const [showSignUpModal, setShowSignUpModal] = useState(false);

	// State to control the visibility of the login modal
	const [showLoginModal, setShowLoginModal] = useState(false);

    // Fetch the logged-in user when the app loads
	useEffect(() => {
		async function fetchLoggedInUser() {
			try {
				const user = await NotesApi.getLoggedInUser();
				setLoggedInUser(user);
			} catch (error) {
				console.error(error);
			}
		}
		fetchLoggedInUser();
	}, []);

	return (
		<BrowserRouter>
			<div>
				{/* Navigation bar component */}
				<NavBar
					loggedInUser={loggedInUser}
					onLoginClicked={() => setShowLoginModal(true)}
					onSignUpClicked={() => setShowSignUpModal(true)}
					onLogoutSuccessful={() => setLoggedInUser(null)}
				/>
				{/* Main content container */}
				<Container className={styles.pageContainer}>
					<Routes>
						{/* Route for the home page (NotesPage) */}
						<Route
							path='/'
							element={<NotesPage loggedInUser={loggedInUser} />}
						/>
						{/* Route for the privacy page */}
						<Route
							path='/privacy'
							element={<PrivacyPage />}
						/>
						{/* Route for any other unknown paths (404 NotFoundPage) */}
						<Route
							path='/*'
							element={<NotFoundPage />}
						/>
					</Routes>
				</Container>
				{/* Sign-up modal */}
				{showSignUpModal &&
					<SignUpModal
						onDismiss={() => setShowSignUpModal(false)}
						onSignUpSuccessful={(user) => {
							setLoggedInUser(user);
							setShowSignUpModal(false);
						}}
					/>
				}
				{/* Login modal */}
				{showLoginModal &&
					<LoginModal
						onDismiss={() => setShowLoginModal(false)}
						onLoginSuccessful={(user) => {
							setLoggedInUser(user);
							setShowLoginModal(false);
						}}
					/>
				}
			</div>
		</BrowserRouter>
	);
}

export default App;
