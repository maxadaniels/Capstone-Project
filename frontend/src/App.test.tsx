import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('App component matches snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
