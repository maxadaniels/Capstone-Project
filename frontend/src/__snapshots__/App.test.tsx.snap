// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-primary sticky-top"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Notes Master
        </a>
        <button
          aria-controls="main-navbar"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="main-navbar"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              href="/privacy"
            >
              Privacy
            </a>
          </div>
          <div
            class="ms-auto navbar-nav"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Sign Up
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Log In
            </button>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="pageContainer container"
    >
      <div
        class="notesPage container"
      >
        <p>
          Please login to see your notes
        </p>
      </div>
    </div>
  </div>
</div>
`;
